src/commands/test/lib/junit-xml-util.ts

```diff
@@ -0,0 +1,174 @@
+import * as pfs from "../../../util/misc/promisfied-fs";
+import { XmlUtil } from "./xml-util";
+import * as fs from "fs";
+import * as path from "path";
+import * as unzip from "unzip";
+import { DOMParser } from "xmldom";
+
+export class JUnitXmlUtil extends XmlUtil {
+
+  async mergeXmlResults(pathToArchive: string): Promise<Document> {
+    let tempPath: string = await pfs.mkTempDir("appcenter-junittestreports")
+    let mainXml: Document = this.getEmptyXmlDocument();
+
+    let self = this;
+    return new Promise<Document>((resolve,reject) => {
```

Comment:
Consider hooking up some 'error' event handlers on these pipes to properly detect if something goes wrong.